packages/launchpad/src/launchpad-internal.tsx

```diff
@@ -1,89 +1,87 @@
-import { recordTracksEvent } from '@automattic/calypso-analytics';
 import { useLaunchpad } from '@automattic/data-stores';
-import { useRef, useMemo, useEffect } from 'react';
-import Checklist from './checklist';
+import { useMemo } from 'react';
+import Checklist, { Placeholder as ChecklistPlaceHolder } from './checklist';
+import ChecklistItem from './checklist-item';
+import { useTracking } from './use-tracking';
 import type { Task } from './types';
-import type { UseLaunchpadOptions } from '@automattic/data-stores';
+import type { SiteDetails, UseLaunchpadOptions } from '@automattic/data-stores';
 
 export interface LaunchpadInternalProps {
+	site?: SiteDetails | null;
+	flow?: string;
 	siteSlug: string | null;
-	checklistSlug?: string | null;
+	checklistSlug: string | null;
 	makeLastTaskPrimaryAction?: boolean;
 	taskFilter?: ( tasks: Task[] ) => Task[];
 	useLaunchpadOptions?: UseLaunchpadOptions;
-	launchpadContext?: string | undefined;
+	launchpadContext: string;
+	// Temporary flag to control the auto tracking of click events
+	enableAutoTracking?: boolean;
 }
 
 /**
  * Low-level launchpad component that should only be used in exceptional cases.
  * Please use the main Launchpad component whenever possible.
  */
 const LaunchpadInternal = ( {
+	flow,
+	site,
 	siteSlug,
 	checklistSlug,
 	taskFilter,
 	makeLastTaskPrimaryAction,
 	useLaunchpadOptions = {},
 	launchpadContext,
+	enableAutoTracking = false,
 }: LaunchpadInternalProps ) => {
 	const launchpadData = useLaunchpad(
 		siteSlug || '',
 		checklistSlug,
 		useLaunchpadOptions,
 		launchpadContext
 	);
+
 	const { isFetchedAfterMount, data } = launchpadData;
 	const { checklist } = data;
-	const tasks = useRef< Task[] >( [] );
-	const tasklistCompleted = checklist?.every( ( task: Task ) => task.completed ) || false;
-
-	const numberOfSteps = checklist?.length || 0;
-	const completedSteps = ( checklist?.filter( ( task: Task ) => task.completed ) || [] ).length;
 
-	useMemo( () => {
-		const originalTasks = data.checklist || [];
-		tasks.current = taskFilter ? taskFilter( originalTasks ) : originalTasks;
-	}, [ data, taskFilter ] );
+	const tasks = useMemo( () => {
+		if ( ! checklist ) {
+			return [];
+		}
+		return taskFilter ? taskFilter( checklist ) : checklist;
+		// Array of tasks requires deep comparison
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ JSON.stringify( checklist ), taskFilter ] );
```

Comment:
Will the `stringify()` output correctly reflect changes/differences in callback functions?

Reply:
Yes, it is better than using the array. 
React compares array just by reference, so the same data but with different instances is considered a new instance and triggers a new render.